mlir/include/mlir/Analysis/Presburger/Fraction.h

```diff
@@ -124,6 +129,9 @@ inline Fraction operator-(const Fraction &x, const Fraction &y) {
   return reduce(Fraction(x.num * y.den - x.den * y.num, x.den * y.den));
 }
 
+// Find the integer nearest to a given fraction.
+inline MPInt round(const Fraction &f) { return floor(f + Fraction(1, 2)); }
```

Comment:
this seems an unnecessarily expensive method. just check remainder rem = num % den and return {num - rem, den} or {num + den - rem, den} appropriately?